### ISSUE TITLE
Code path analysis for (a && b) ?? c

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#13614]
**Tell us about your environment**

* **ESLint Version:** v7.7.0
* **Node Version:** v12.14.0
* **npm Version:** 6.13.4

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  parserOptions: {
    ecmaVersion: 2020
  }
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
(a && b) ?? c;
```

The rest of the bug report template isn't quite suitable for this issue.

Code path analysis gives the following:

<details>
<summary>DOT</summary>

```dot
digraph {
node[shape=box,style="rounded,filled",fillcolor=white];
initial[label="",shape=circle,style=filled,fillcolor=black,width=0.25,height=0.25];
final[label="",shape=doublecircle,style=filled,fillcolor=black,width=0.25,height=0.25];
s1_1[label="Program:enter\nExpressionStatement:enter\nLogicalExpression:enter\nLogicalExpression:enter\nIdentifier (a)"];
s1_2[label="Identifier (b)\nLogicalExpression:exit"];
s1_3[label="Identifier (c)"];
s1_4[label="LogicalExpression:exit\nExpressionStatement:exit\nProgram:exit"];
initial->s1_1->s1_2->s1_3->s1_4;
s1_1->s1_4;
s1_2->s1_4->final;
}
```
</details>

<details>
<summary>Image</summary>

![image](https://user-images.githubusercontent.com/44349756/91108055-31d8a600-e677-11ea-89b8-801b95d2c6af.png)

</details>

This would be correct for `(a || b) ?? c`, but I think it isn't entirely correct for `(a && b) ?? c` since there should be a path from `a` to `c` that doesn't go through `b`:

```js
(null && console.log("b")) ?? console.log("c"); // logs only "c"
```

**Comment by btmills** [comment]
This is a similar issue to #13634. I believe we can fix both in semver-minor changes.

**Comment by aladdin-add** [comment]
friendly ping @mdjermanovic is this still an issue we want to address?

**Comment by nzakas** [comment]
Yes, we should fix this.

**Comment by nzakas** [comment]
Working on this.
